[Ong Chong Yao] iP

chongyaoo opened the pull request.

chongyaoo committed:
Update Grizzly.java

chongyaoo committed:
Add add, list, mark and unmark functions

Add add function to the chatbot:
add items to list of items to do
Add list function:
list the items to do
Add mark function:
mark items done
Add unmark function:
unmark items not done

chongyaoo committed:
Update Grizzly.java

Implemented some changes to the logic of chatbot

chongyaoo committed:
Add Task class

Improve logic flow of chatbot
Implement Classes Deadline, Todo and Event that inherit from Class Task following OOP

msc-123456 commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+
+    @Override
+    public String toType() {
+        return "[D]";
```
I think you can type this like "return "[D]" + super.toString() + " (by: " + by + ")";"?

msc-123456 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+        String logo = """
+                Hi, I'm your friendly assistant, Grizzly!
+                What can I do for you today?
+                ____________________________________________________________""";
```
maybe you can add more lines to seperate each action?

msc-123456 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+            line = in.nextLine();
+        }
+    }
+}
```
this part is a little complex, I think it's better to use functions to handle different situations

ChangIkJoong replied:
This part I would suggest to abstract into a class perhaps handling the output, the other the logic. 
Right now it seems a bit overly complex and hard to follow.

ChangIkJoong replied:
This part I would suggest to abstract into a class perhaps handling the output, the other the logic.
Right now it seems a bit overly complex and hard to follow.

msc-123456 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+public class Grizzly {
+    public static void main(String[] args) {
+        String logo = """
+                Hi, I'm your friendly assistant, Grizzly!
```
maybe you can add a logo like the given "Duke"

msc-123456 reviewed (commented):
it's good to use separate files to make todo、event and deadline, but the main part is too complex

ChangIkJoong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        } else {
+            return "[ ] ";
+        }
+    }
```
I would keep the output of let's say a string seperate and just keep this as a boolean within the class, then another class perhaps handling the string output.

ChangIkJoong commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+                        storedItems[itemToMark - 1].setMarked(true);
+                        System.out.println("Nice! I've marked this task as done:");
+                        System.out.println(itemToMark + "." + storedItems[itemToMark - 1].toType() + storedItems[itemToMark - 1].MarkedBox() + storedItems[itemToMark - 1].getString());
+                    } else {
```
Feels like this is a bit too nestled, perhaps the if function can be abstracted to it's own method for improved readability?

ChangIkJoong commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+public class Task {
+    protected String data;
```
Maybe the name data can be a bit too open, doesn't describe too much what kind of data or what it is.

ChangIkJoong reviewed (commented):
Overall nice coding, I like the inheritance. I would suggest to abstract some methods and increase readability for certain parts as provided, as some parts had perhaps a bit too deep nesting and overall a bit hard to read some functional aspects as they became quite long.

JianHao24 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+        line = in.nextLine();
+
+
+        while (true) {
```
The code can be refactored by extracting the code in each cases, this would promote better abstraction and support readability.

JianHao24 commented on src/main/java/Task.java:
```diff
@@ -0,0 +1,41 @@
+        this.isMarked = marked;
+    }
+
+    public String MarkedBox() {
```
method MarkedBox() should follow lowerCamelCase, perhaps rename as markedBox() or getMarkedBox() since it returns a value.

JianHao24 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+        Scanner in = new Scanner(System.in);
+        String line;
+        int index = 1;
+        Task[] storedItems = new Task[100];
```
Can use a constant for the maximum array size instead of an arbitrary number 100 for better readability.  Maybe define a static final MAX_TASKS = 100 at the start.

JianHao24 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,97 @@
+                    try {
+                        itemToUnmark = Integer.parseInt(parts[1]);
+                    } catch (NumberFormatException e) {
+                        System.out.println("Not a valid integer.");
```
Maybe you can define your errors as constants at the start, maybe naming it something like INVALID_INT_ERROR. This promotes readability of code and also reduces redundancy should the error message be reused. The constant also helps standardise error message in case one accidentally varies it.

JianHao24 commented on src/main/java/ToDo.java:
```diff
@@ -0,0 +1,12 @@
+    }
+
+    @Override
+    public String toType() {
```
You can overload the toString function to print the boxes which mark and represent class type when printing the instance instead of using a seperate toType function.

chongyaoo committed:
Update Task classes

Update the Event and Deadline inherited classes to contain the string for dates
Write test cases in input.txt and runtest.bash to check the outputs of the chatbot

chongyaoo committed:
Separate chatbot cases into functions

Create void functions which handle different inputs separately, separating functionalities code from the main function

chongyaoo committed:
Handle Errors

Add checks for errors like incorrect command input, and handle these cases

syCHEN1645 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,150 @@
+            String[] parts = line.split("\\s+", 2);
+            String action = parts[0].toLowerCase();
+            switch (action) {
+                case "bye":
```
Maybe consider refactor magic literals with final strings?

syCHEN1645 commented on src/main/java/Grizzly.java:
```diff
@@ -0,0 +1,150 @@
+                    try {
+                        itemToMark = Integer.parseInt(parts[1]);
+                    } catch (NumberFormatException e) {
+                        System.out.println("Not a valid integer.");
```
This message seems to be used in other cases too, maybe define a constant variable for it?

syCHEN1645 reviewed (commented):
Good effort! Your code looks clear, just a bit too bulky. It would be graeat if you can break it down to smaller methods!

chongyaoo committed:
Update Grizzly.java

chongyaoo committed:
Add Delete functionality

Add handleDelete() function
Use Java Collections class for storing data with dynamic array

chongyaoo committed:
Merge branch 'heads/branch-level-5'

chongyaoo committed:
Add delete and Save functions

Delete function removes Task from dynamic list
Save function saves to lines.txt under src whenever exiting, ensuring that you can reload list of Tasks when starting the program again

chongyaoo committed:
Level-6-branch

chongyaoo committed:
Branch-Level-7

chongyaoo committed:
Merge branch 'branch-level-7'

chongyaoo committed:
More OOP

Extract out Parser, Storage, TaskList and Ui classes to carry out functionalities of chatbot

chongyaoo committed:
Add Find functionality

Allows user to search for a task in the array

chongyaoo committed:
Further commits

chongyaoo committed:
Merge pull request #2 from chongyaoo/level-9-branch

Merging Level 9 branch into master branch

chongyaoo committed:
Add JavaDoc comments to code

Add JavaDoc description to Classes and methods

chongyaoo committed:
Merge pull request #3 from chongyaoo/branch-A-JavaDoc

Add JavaDoc comments to code

chongyaoo committed:
Update README

chongyaoo committed:
Update README

Update README, as well as input and expected.txt

chongyaoo committed:
Edit Storage class

chongyaoo committed:
Update JavaDoc comments

Add more details to JavaDoc comments
